### DISCUSSION TITLE
Technical Debt Busting

### DISCUSSION THREAD

**Discussion by louwers** [maplibre/maplibre#280] [Cross-project feature proposals]
This bounty direction feels like it is missing.

We have great ambitions for the MapLibre projects. Amongst other things, we want to have top-notch documentation, we want to support more writing systems, have a globe view and have better 3D support. In general, we MapLibre to be the best and most versatile vector-tile based maps toolkit available. However, this will not be possible if there are technical challenges holding us back. I am talking of course about technical debt.

> Software systems are prone to the build up of cruft - deficiencies in internal quality that make it harder than it would ideally be to modify and extend the system further. Technical Debt is a metaphor, coined by Ward Cunningham, that frames how to think about dealing with this cruft, thinking of it like a financial debt. The extra effort that it takes to add new features is the interest paid on the debt.

Source: [Martin Fowler](https://martinfowler.com/bliki/TechnicalDebt.html)

Another reason why it is important to have this bounty direction is that it is not always the most fun work to do resolve technical debt. Along with the warm fuzzy feeling of making your favourite maps toolkit better, a financial incentive would likely help to get these kinds of issues resolved.

This bounty direction is not about modernization, which may be another good bounty direction, but explicitly about resolving technical challenges that are actively holding us back from achieving some other goal.

**Comment by HarelM** [comment]
While I don't argue that tech debt is important to fix/resolve, I think that adding yet another bounty, with even more ambiguity on what it actually means it's not the correct approach.
We have bounties on docs, on performance, on modernization. I'm pretty sure you could find one of those as the right bounty direction.
If tech debt is a pre requisite for one of the above then you can have a bounty for the tech debt in the direction of the "carrying" feature/issue/bug.

Tech debt is so general from my point of view that you might as well say: everything can be considered as a reason for a bounty, let's drop the concept of bounty directions.

**Reply by louwers** [reply]
I didn't know the Modernization bounty was approved!

I think this would be a good bounty direction to pin it under.
